Updates to the sign up form design and account creation copy

sandymcfadden opened the pull request:
### Fix

With the new sign up flow, there were some copy changes and form design changes that needed to be updated. This updates those.

**Before Screenshots**

**Light mode**

<details>
<summary>Initial form</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068610-7ca8d600-9874-11eb-8da7-4eda5ed3c065.png" />
</details>
<details>
<summary>Initial form with focus</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068638-8599a780-9874-11eb-97eb-799cd30b6ef1.png" />
</details>
<details>
<summary>Form with email and focus</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068685-8e8a7900-9874-11eb-90d3-8aca56500866.png" />
</details>
<details>
<summary>Form with email</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068712-93e7c380-9874-11eb-9b68-fa29d5a52246.png" />
</details>
<details>
<summary>Next steps message</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068755-9c3ffe80-9874-11eb-9b12-1f3c6f078650.png" />
</details>

**Dark mode**

<details>
<summary>Initial form</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068807-acf07480-9874-11eb-8f98-be63e97159b9.png" />
</details>
<details>
<summary>Initial form with focus</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068830-b2e65580-9874-11eb-9501-0165a2813534.png" />
</details>
<details>
<summary>Form with email and focus</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068850-bd085400-9874-11eb-900c-cf61c63a61b0.png" />
</details>
<details>
<summary>Form with email</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068885-c691bc00-9874-11eb-92a1-672c8cd5f23c.png" />
</details>
<details>
<summary>Next steps message</summary>
<img src="https://user-images.githubusercontent.com/1326294/114068924-d1e4e780-9874-11eb-8f4e-62c7ad0d1ca7.png" />
</details>

**After Screenshots**

**Light mode**

<details>
<summary>Initial form</summary>
<img src="https://user-images.githubusercontent.com/1326294/114065468-14a4c080-9871-11eb-8121-545bc1396dce.png" />
</details>
<details>
<summary>Initial form with focus</summary>
<img src="https://user-images.githubusercontent.com/1326294/114065544-29815400-9871-11eb-81b1-516c16ac2263.png" />
</details>
<details>
<summary>Form with email and focus</summary>
<img src="https://user-images.githubusercontent.com/1326294/114065628-40c04180-9871-11eb-80d8-33863f1341e3.png" />
</details>
<details>
<summary>Form with email</summary>
<img src="https://user-images.githubusercontent.com/1326294/114065689-4f0e5d80-9871-11eb-8d84-72ff7d583700.png" />
</details>
<details>
<summary>Next steps message</summary>
<img src="https://user-images.githubusercontent.com/1326294/114065711-57669880-9871-11eb-87b0-f5015834b9d8.png" />
</details>

**Dark mode**

<details>
<summary>Initial form</summary>
<img src="https://user-images.githubusercontent.com/1326294/114066966-a82ac100-9872-11eb-84b7-e0bc218323c5.png" />
</details>
<details>
<summary>Initial form with focus</summary>
<img src="https://user-images.githubusercontent.com/1326294/114067038-bd075480-9872-11eb-89a5-4f115a3e278a.png" />
</details>
<details>
<summary>Form with email and focus</summary>
<img src="https://user-images.githubusercontent.com/1326294/114067094-d01a2480-9872-11eb-868d-7a42d10ae547.png" />
</details>
<details>
<summary>Form with email</summary>
<img src="https://user-images.githubusercontent.com/1326294/114067116-d6100580-9872-11eb-9cc3-35c650f79878.png" />
</details>
<details>
<summary>Next steps message</summary>
<img src="https://user-images.githubusercontent.com/1326294/114067165-e1fbc780-9872-11eb-9b7b-38a272b7eb60.png" />
</details>

Resolves #2823

### Test

Test in both light and dark mode.

1. Ensure you are not logged in.
2. Choose to sign up for an account.
3. Ensure the form matches the designs.
4. Ensure the copy after requesting matches the updated copy.

### Release

- Updated the design of the login/sign up form as well as clarifying the message shown when requesting an account.

sandymcfadden committed:
Update sign up copy

sandymcfadden committed:
Update styles for login/signup inputs

sandymcfadden committed:
Add in readonly styles

sandymcfadden committed:
Add dark mode read only styles

SylvesterWilmott commented:
Looks good @sandymcfadden 

Can we add "nowrap" to the "Terms of Service" link or maybe a line break before it so it doesn't wrap mid-link?

When focusing the fields, the text does a small jump, is this because of the increase in border size? Maybe using a box shadow for the focus border would be better in this case (similar to the `validate:invalid` style).

fluiddot commented on lib/auth/style.scss:
```diff
@@ -194,4 +210,22 @@
+    border-color: $studio-gray-60;
+
+    &:focus {
+      border: 2px solid $studio-simplenote-blue-30;
+      outline: none;
```
I was wondering if since some `border` properties are already defined in the default theme (light), here we could only have the border color as we have in the other states (disabled and read-only):
`border-color: $studio-simplenote-blue-30;`

And the same for `outline`, I think we could remove it from here too.

sandymcfadden replied:
Thanks! You're right that is a good call. Based on some other feedback I've switched to using box-shadow instead of border, and I don't believe there is a way to change only the color.
I did drop the outline though.

fluiddot reviewed (commented):
I tested it and looks great 🎊 , I added just a minor comment.

sandymcfadden committed:
Stop terms of service link from wrapping to new line

sandymcfadden committed:
Switch to using box-shadow instead of border to prevent shifting of content

sandymcfadden commented:
Thanks @SylvesterWilmott!
> Can we add "nowrap" to the "Terms of Service" link or maybe a line break before it so it doesn't wrap mid-link?
I've updated this now to fix the wrapping. 
 
> When focusing the fields, the text does a small jump, is this because of the increase in border size? Maybe using a box shadow for the focus border would be better in this case (similar to the `validate:invalid` style).

You're right it was because of the change in border size. I've updated it to use a box-shadow now instead as you suggested.

fluiddot commented on lib/auth/style.scss:
```diff
@@ -63,6 +64,22 @@
   }
 
+  input:focus {
+    border-color: transparent;
+    box-shadow: 0 0 0 2px $studio-simplenote-blue-50;
```
I saw that the border is maybe expanded too much when the element is focused (see attached screenshot), does this match the designs? (I can't check them because I don't have access 😄 )
<img width="697" alt="Screenshot 2021-04-13 at 11 16 08" src="https://user-images.githubusercontent.com/14905380/114529064-e09f1600-9c49-11eb-946d-3dd0ccb1c50a.png">

I was wondering if we could set the border color with the same blue and instead of having two pixels in the box shadow, reduce it to one (see attached screenshot):
```
border-color: $studio-simplenote-blue-50;
box-shadow: 0 0 0 1px $studio-simplenote-blue-50;
```

<img width="684" alt="Screenshot 2021-04-13 at 11 19 33" src="https://user-images.githubusercontent.com/14905380/114529474-44c1da00-9c4a-11eb-97e1-171c6f89ceda.png">

<hr>

Additionally I'm not sure how the element should look like when it's invalid and focused, now the border is set to grey but the focus color should be blue, is this right?

Here you can check how it looks like with the current changes and what I proposed:
Current changes|Proposal
--|--
<video src=https://user-images.githubusercontent.com/14905380/114528023-e516ff00-9c48-11eb-99a8-66f2fa465564.mp4>|<video src=https://user-images.githubusercontent.com/14905380/114528110-fb24bf80-9c48-11eb-85c4-f3e1e908094e.mp4>

sandymcfadden replied:
I like the suggestion of going with both border and box-shadow. I'm going to work on that. I have screenshots of the designs in the issue related to this #2823 under the light and dark mode toggles. 

@SylvesterWilmott how would you want the invalid state to look when focused? I had left it as it was previously, but we could update it for sure.

SylvesterWilmott replied:
>I had left it as it was previously, but we could update it for sure.

As it is currently looks good to me!

fluiddot commented on lib/auth/style.scss:
```diff
@@ -194,4 +212,21 @@
+    border-color: $studio-gray-60;
+
+    &:focus {
+      box-shadow: 0 0 0 2px $studio-simplenote-blue-30;
```
I think we have to add here `border-color: transparent;`, otherwise the dark theme sets the border color to grey when it's focused as per line 216.

Here is how it looks like currently, observe the thin inner grey line:
<img width="1192" alt="Screenshot 2021-04-13 at 11 28 16" src="https://user-images.githubusercontent.com/14905380/114530769-8d2dc780-9c4b-11eb-8196-1aa987746533.png">

And here are the styles of the email input when it's focused, observe that `.theme-dark .login input` sets the border color to grey:
<img width="313" alt="Screenshot 2021-04-13 at 11 29 34" src="https://user-images.githubusercontent.com/14905380/114530816-96b72f80-9c4b-11eb-8110-576d2bc66269.png">

sandymcfadden replied:
Thanks for catching this. 
I think I'll solve this by going with the recommendation in your other comment to use a combination of border and box-shadow.

sandymcfadden committed:
Update focus borders to use a combination of border and box-shadow

fluiddot commented on lib/auth/style.scss:
```diff
@@ -63,6 +64,22 @@
+    color: $studio-gray-20;
+  }
+
+  input:read-only {
+    border-color: $studio-gray-10;
+    background-color: rgba(0, 0, 0, 0.05);
+  }
```
I saw that it's possible to have both `focus` and `read-only` states applied at the same time. In this case the `read-only` styles have preference over the `focus` ones because they're located after and therefore the inner border is set to grey. 

If we want to keep the same styles (only have blue border), we should put this part before `input:focus` and the same in the code below for the dark theme.

Here you can check how it looks like when a `readonly` input is focused:
<img width="1077" alt="Screenshot 2021-04-14 at 10 05 52" src="https://user-images.githubusercontent.com/14905380/114676426-92514c00-9d09-11eb-9dcd-e6be947215e9.png">

sandymcfadden replied:
Another good catch, thank you! I've updated this appropriately now.

sandymcfadden committed:
Changing order of CSS rules so focus styles will be properly applied to readonly inputs

fluiddot reviewed (approved):
LGTM 🎊 !

SylvesterWilmott reviewed (approved):
Looks good! Thanks @sandymcfadden
